lib/hangman2.rb

```diff
@@ -1,39 +1,68 @@
 class Word
   def initialize(word)
     @word = word
+    @word_as_letter = word.split('')
     @user_guesses = []
+    @lives = 6
   end
 
-class Life
-  def amount_of_lives(life)
-    @life = life
-    @amount_of_lives = 6
+
+  def amount_of_lives
+    @lives
   end
 
+  # def final_result
+  #     if @lives >= 1
+  #     letter.join == @word
+  #       #letters.join
+  #     p "bravo! you won"
+  #   end
+  # end
+
+
 
-# ASKING FOR A LETTER
-    def add_guess(letter)
-      @user_guesses.push(letter)
+  ##-------REQUEST
+  def add_guess(letter)
+    @user_guesses.push(letter)
+    if @word_as_letter.include?(letter)
+      right_guess()
+    elsif
+      wrong_guess()
+      #p "You start with #{@lives} lives. "
     end
+  end
+
+
 
 
 
 
-# DEFINING THE RIGHT AND WRONG GUESSES
-    def guessed_letters
-      letters = []
-      @word.split('').each do |letter|
-        if @user_guesses.include?(letter)
-          letters.push(letter)
-          # it's counting the amount of empty '_'
-          # puts "Yay almost done, give a new letter!"
-        else @user_guesses != @word
-          letters.push('_')
-          #@amount_of_lives - 1
-          puts "you loose one life"
-        end
+  def right_guess
+    letters = []
+    @word_as_letter.each do |letter|
+      if
+        @user_guesses.include?(letter)
+        letters.push(letter)
+      else
+        letters.push('_')
       end
-      letters
+    end
+    letters
+    # || final_result = (letters.join == @word) && (@lives >= 1) # && (@word_as_letter.join == @word)
+     p "You guess right! Give me another letter :" || "great! you won"
+  end
+
+
+
+
+  def wrong_guess
+    if @lives >= 1
```

Comment:
This can just be `if @lives > 1`
Also, maybe `@count_of_lives`, or `@number_or_lives` would describe better what the value of this variable is (= a number)?